python/tests/test_integration_schema.py

```diff
@@ -0,0 +1,2617 @@
+    tbl = _create_table_with_schema(
+        catalog,
+        Schema(
+            NestedField(field_id=1, name="bar", field_type=IntegerType(), required=True),
+            NestedField(
+                field_id=2,
+                name="location",
+                field_type=ListType(
+                    type="list",
+                    element_id=3,
+                    element_type=StructType(
+                        NestedField(field_id=4, name="latitude", field_type=FloatType(), required=False),
+                        NestedField(field_id=5, name="longitude", field_type=FloatType(), required=False),
+                    ),
+                    element_required=True,
+                ),
+                required=True,
+            ),
+        ),
+    )
+
+    with tbl.update_schema(case_sensitive=False) as schema_update:
+        schema_update.update_column("baR", LongType())
+        schema_update.update_column("Location.Latitude", DoubleType())
+        schema_update.update_column("Location.Longitude", DoubleType())
+
+    assert tbl.schema() == Schema(
+        NestedField(field_id=1, name="bar", field_type=LongType(), required=True),
+        NestedField(
+            field_id=2,
+            name="location",
+            field_type=ListType(
+                type="list",
+                element_id=3,
+                element_type=StructType(
+                    NestedField(field_id=4, name="latitude", field_type=DoubleType(), required=False),
+                    NestedField(field_id=5, name="longitude", field_type=DoubleType(), required=False),
+                ),
+                element_required=True,
+            ),
+            required=True,
+        ),
+    )
+
+
+allowed_promotions = [
+    (StringType(), BinaryType()),
+    (BinaryType(), StringType()),
+    (IntegerType(), LongType()),
+    (FloatType(), DoubleType()),
+    (DecimalType(9, 2), DecimalType(18, 2)),
+]
+
+
+@pytest.mark.parametrize("from_type, to_type", allowed_promotions, ids=str)
+@pytest.mark.integration
+def test_allowed_updates(from_type: PrimitiveType, to_type: PrimitiveType, catalog: Catalog) -> None:
+    tbl = _create_table_with_schema(
+        catalog,
+        Schema(
+            NestedField(field_id=1, name="bar", field_type=from_type, required=True),
+        ),
+    )
+
+    with tbl.update_schema() as schema_update:
+        schema_update.update_column("bar", to_type)
+
+    assert tbl.schema() == Schema(NestedField(field_id=1, name="bar", field_type=to_type, required=True))
+
+
+disallowed_promotions_types = [
+    BooleanType(),
+    IntegerType(),
+    LongType(),
+    FloatType(),
+    DoubleType(),
+    DateType(),
+    TimeType(),
+    TimestampType(),
+    TimestamptzType(),
+    StringType(),
+    UUIDType(),
+    BinaryType(),
+    FixedType(3),
+    FixedType(4),
+    # We'll just allow Decimal promotions right now
+    # https://github.com/apache/iceberg/issues/8389
```

Comment:
Changing the scale of a decimal is not an allowed promotion.

Reply:
Yes it passes right now :)